feat: add kubeshark/kubeshark

Chemsmith opened the pull request:
[kubeshark/kubeshark](https://github.com/kubeshark/kubeshark): The API traffic viewer for Kubernetes providing real-time, protocol-aware visibility into Kubernetes’ internal network, capturing, dissecting and monitoring all traffic and payloads going in, out and across containers, pods, nodes and clusters. Think TCPDump and Wireshark re-invented for Kubernetes

```console
$ aqua g -i kubeshark/kubeshark
```

## How to confirm if this package works well

Reviewers aren't necessarily familiar with this package, so please describe how to confirm if this package works well. Please confirm if this package works well yourself as much as possible.

Command and output

```console
$ kubeshark --help
Kubeshark: The API Traffic Viewer for Kubernetes
An extensible Kubernetes-aware network sniffer and kernel tracer.
For more info: https://kubeshark.co

Usage:
  kubeshark [command]

Available Commands:
  check       Check the Kubeshark resources for potential problems
  clean       Removes all Kubeshark resources
  completion  Generate the autocompletion script for the specified shell
  config      Generate Kubeshark config with default values
  help        Help about any command
  logs        Create a ZIP file with logs for GitHub issues or troubleshooting
  proxy       Open the web UI (front-end) in the browser via proxy/port-forward.
  tap         Capture the network traffic in your Kubernetes cluster.
  version     Print version info

Flags:
      --configpath string   Override config file path using --configpath (default "/home/<user>/.kubeshark/config.yaml")
  -d, --debug               Enable debug mode.
  -h, --help                help for kubeshark
      --set strings         Override values using --set

Use "kubeshark [command] --help" for more information about a command.
```

Chemsmith committed:
feat: add kubeshark/kubeshark

[kubeshark/kubeshark](https://github.com/kubeshark/kubeshark): The API traffic viewer for Kubernetes providing real-time, protocol-aware visibility into Kubernetes’ internal network, capturing, dissecting and monitoring all traffic and payloads going in, out and across containers, pods, nodes and clusters. Think TCPDump and Wireshark re-invented for Kubernetes

```console
$ aqua g -i kubeshark/kubeshark
```

Reviewers aren't necessarily familiar with this package, so please describe how to confirm if this package works well.
Please confirm if this package works well yourself as much as possible.

Command and output

```console
$
```

If files such as configuration file are needed, please share them.

```
```

Reference

-

CrystalMethod commented on pkgs/kubeshark/kubeshark/registry.yaml:
```diff
@@ -0,0 +1,21 @@
+        asset: kubeshark.exe
+    checksum:
+      type: github_release
+      asset: kubeshark_{{.OS}}_{{.Arch}}.sha256
```
You can use shorter `asset: "{{.Asset}}.sha256"` which is an alias to definition of `asset` in line 5

CrystalMethod commented on pkgs/kubeshark/kubeshark/registry.yaml:
```diff
@@ -0,0 +1,21 @@
+    checksum:
+      type: github_release
+      asset: kubeshark_{{.OS}}_{{.Arch}}.sha256
+      file_format: regexp
```
`file_format: raw` is better here since the checksum file only contains the sha

suzuki-shunsuke replied:
This checksum file contains not only sha but also asset names.

```console
$ curl -L https://github.com/kubeshark/kubeshark/releases/download/38.5/kubeshark_38.5_checksums.txt
1507fca5aa4063ab31d0a8f8dc4916a4d03defa250e6ed190139c7cd7439b6fa  kubeshark_38.5_windows_arm64.tar.gz
15777e59512b5a85417c8ab7d7e4a8e65166c94773fab7466118be53d7f72fb3  kubeshark_38.5_windows_386.tar.gz
386b62c1fe042acf6d04e9c7be10c87e406cafbf72ef39f1cc6245eb1332fbb1  kubeshark_38.5_linux_amd64.tar.gz
63dfb7029466fb01f6858fda31357966bc0b2a4f7b1ad652204e92ac835327da  kubeshark_38.5_linux_386.tar.gz
7ed6e290eee3a9ca3776e41eee4033b34aa8e9fa3f66fa374df567e2d7efe44a  kubeshark_38.5_darwin_arm64.tar.gz
88193e5c1c73de2daa840e0a954753a719f15fb8c23d748d626c19e339f72dd6  kubeshark_38.5_linux_arm64.tar.gz
89bf5e7953f4c5774a27da6ebd4c07a3e37b56e5ce7feaca54b1c8000f7c5f23  kubeshark_38.5_windows_amd64.tar.gz
8b68f6de3152ce8a3952e9beafa4c05e83832321ec004966f7067899b0209790  kubeshark_38.5_darwin_amd64.tar.gz
```

suzuki-shunsuke replied:
Oh, sorry. I misunderstood.

suzuki-shunsuke replied:
We should use `kubeshark_38.5_checksums.txt` rather than `kubeshark_{{.OS}}_{{.Arch}}.sha256` because we can save API call.

CrystalMethod commented on pkgs/kubeshark/kubeshark/registry.yaml:
```diff
@@ -0,0 +1,21 @@
+      asset: kubeshark_{{.OS}}_{{.Arch}}.sha256
+      file_format: regexp
+      algorithm: sha256
+      pattern:
```
You can omit `pattern` and `checksum` if `file_format: raw` is used.

CrystalMethod commented on pkgs/kubeshark/kubeshark/registry.yaml:
```diff
@@ -0,0 +1,21 @@
+      - windows/amd64
+      - darwin
+      - linux
+    overrides:
```
Please add `format: raw` since the asset is not an archive.

suzuki-shunsuke replied:
kubeshark releases both `raw` and `tar.gz`. https://github.com/kubeshark/kubeshark/releases/tag/38.5
I prefer `tar.gz` than `raw` because the file size is small.

CrystalMethod commented on pkgs/kubeshark/kubeshark/registry.yaml:
```diff
@@ -0,0 +1,21 @@
+    repo_owner: kubeshark
+    repo_name: kubeshark
+    asset: kubeshark_{{.OS}}_{{.Arch}}
+    description: The API traffic viewer for Kubernetes providing real-time, protocol-aware visibility into Kubernetes internal network capturing, dissecting and monitoring all traffic and payloads going in, out and across containers, pods, nodes and clusters.
```
Please shorten it to e.g. `The API traffic viewer for Kubernetes`

CrystalMethod reviewed (commented):
I myself am not yet that experienced with the Aqua package definition and regularly make mistakes. So please do not misunderstand my comments.

suzuki-shunsuke commented:
Thank you for your contribution!

suzuki-shunsuke commented:
@CrystalMethod Thank you for your review!

suzuki-shunsuke commented:
Let me add some commits.

suzuki-shunsuke committed:
fix(kubeshark/kubeshark): shorten the description

suzuki-shunsuke committed:
refactor(kubeshark/kubeshark): refactor supported_envs and overrides

suzuki-shunsuke committed:
fix(kubeshark/kubeshark): use tar.gz instead of raw format

kubeshark releases both raw and tar.gz. https://github.com/kubeshark/kubeshark/releases/tag/38.5
I prefer tar.gz than raw because the file size is small.

suzuki-shunsuke commented:
Was `kubeshark` formely `mizu`?
https://github.com/kubeshark/kubeshark/releases/tag/36.0

suzuki-shunsuke commented:
https://github.com/up9inc/mizu is redirected to https://github.com/kubeshark/kubeshark .
And aqua-registry already contains [up9inc/mizu](https://github.com/aquaproj/aqua-registry/tree/main/pkgs/up9inc/mizu).

So we should merge up9inc/mizu to kubeshark/kubeshark.

suzuki-shunsuke committed:
fix(kubeshark/kubeshark): merge up9inc/mizu to kubeshark/kubeshark

suzuki-shunsuke committed:
fix(kubeshark/kubeshark): fix checksum config

suzuki-shunsuke commented:
> https://github.com/up9inc/mizu is redirected to https://github.com/kubeshark/kubeshark . And aqua-registry already contains [up9inc/mizu](https://github.com/aquaproj/aqua-registry/tree/main/pkgs/up9inc/mizu).
> 
> So we should merge up9inc/mizu to kubeshark/kubeshark.
> 
> ref. https://aquaproj.github.io/docs/reference/registry-config/aliases

Done.

Chemsmith commented:
@suzuki-shunsuke

Thanks for the assistance on this PR. In future I'd be happy to put in the required work myself with just some pointers to the relevant documentation.

suzuki-shunsuke commented:
Thank you. We always welcome your contribution.
And I'm sorry for preventing you from fixing this pr yourself.
I'll take care in the next time.

suzuki-shunsuke commented:
Released. https://github.com/aquaproj/aqua-registry/releases/tag/v3.132.0
